src/lib/containers/download_list/DownloadListTable.tsx

```diff
@@ -204,28 +218,99 @@ export default function DownloadListTable(props: DownloadListTableProps) {
     }
   }
 
+  const sortColumn = async (column: string) => {
+    try {
+      setIsLoading(true)
+
```

Comment:
Instead of `setColumns` you would have this

```tsx
// store isAscending to be passed into sort function below
const isAscending = column === sortedColumn.column ? !sortedColumn.isAscending: true

setSortedColumn({
   column,
   isAscending
})
```

The `column` can be set to the column no matter what the value is given, doesn't matter if it's setting the same value or if its a new one, but for is`Descending` we need extra steps. If the column coming in was the last column set, then swap the sort direction (e.g. if they click the sort button twice for one column) .

Otherwise, we sort ascendingly for the new column that is being sorted.